File: generate_impostor_data.py
import re
import os

MD_FILE = os.path.join(os.path.dirname(__file__), 'remixed-662bd0ea.md')
OUT_DIR = os.path.join(os.path.dirname(__file__), 'lib', 'core', 'constants')

if not os.path.exists(OUT_DIR):
    os.makedirs(OUT_DIR)

with open(MD_FILE, 'r', encoding='utf-8') as f:
    content = f.read()

def escape_dart(s):
    return s.replace('\\', '\\\\').replace("'", "\\'")

# Find all categories
category_matches = list(re.finditer(r'# .*CATEGORY \d+: (.*)', content))
categories = []

for i, match in enumerate(category_matches):
    cat_title = match.group(1).strip()
    
    start_idx = match.end()
    end_idx = category_matches[i+1].start() if i + 1 < len(category_matches) else len(content)
    
    block = content[start_idx:end_idx]
    
    # Extract numbered lines
    words = []
    for line in block.splitlines():
        line = line.strip()
        m = re.match(r'^\d+\.\s+(.*)', line)
        if m:
            words.append(m.group(1).strip())
            
    if words:
        categories.append((cat_title, words))

print(f"Found {len(categories)} categories.")

dart_code = "/// Impostor Mode word database\n"
dart_code += "class ImpostorData {\n"
dart_code += "  static const Map<String, List<String>> categories = {\n"

for cat_title, words in categories:
    dart_code += f"    '{escape_dart(cat_title)}': [\n"
    for w in words:
        dart_code += f"      '{escape_dart(w)}',\n"
    dart_code += "    ],\n"
    print(f"  {cat_title}: {len(words)} words")

dart_code += "  };\n"
dart_code += "}\n"

with open(os.path.join(OUT_DIR, 'impostor_data.dart'), 'w', encoding='utf-8') as f:
    f.write(dart_code)

print("✅ impostor_data.dart generated successfully!")

File: generate_impostor_data.dart
import 'dart:io';

void main() {
  final mdFile = File('remixed-662bd0ea.md');
  final outDir = Directory('lib/core/constants');
  
  if (!outDir.existsSync()) {
    outDir.createSync(recursive: true);
  }

  final content = mdFile.readAsStringSync();
  final lines = content.split('\n');

  final categories = <String, List<String>>{};
  String? currentCategory;

  final categoryRegex = RegExp(r'^# .*CATEGORY \d+: (.*)$');
  final wordRegex = RegExp(r'^\d+\.\s+(.*)$');

  for (final line in lines) {
    final catMatch = categoryRegex.firstMatch(line.trim());
    if (catMatch != null) {
      currentCategory = catMatch.group(1)!.trim();
      categories[currentCategory] = [];
      continue;
    }

    if (currentCategory != null) {
      final wordMatch = wordRegex.firstMatch(line.trim());
      if (wordMatch != null) {
        categories[currentCategory]!.add(wordMatch.group(1)!.trim());
      }
    }
  }

  String escapeDart(String s) {
    return s.replaceAll('\\', '\\\\').replaceAll("'", "\\'");
  }

  final sb = StringBuffer();
  sb.writeln('/// Impostor Mode word database');
  sb.writeln('class ImpostorData {');
  sb.writeln('  static const Map<String, List<String>> categories = {');

  for (final entry in categories.entries) {
    if (entry.value.isEmpty) continue;
    sb.writeln("    '${escapeDart(entry.key)}': [");
    for (final word in entry.value) {
      sb.writeln("      '${escapeDart(word)}',");
    }
    sb.writeln("    ],");
    print('  ${entry.key}: ${entry.value.length} words');
  }

  sb.writeln('  };');
  sb.writeln('}');

  File('lib/core/constants/impostor_data.dart').writeAsStringSync(sb.toString());
  print('✅ impostor_data.dart generated successfully!');
}
